pkg/instancetype/instancetype.go

```diff
@@ -1386,6 +1387,41 @@ func applyInputPreferences(preferenceSpec *instancetypev1beta1.VirtualMachinePre
 	}
 }
 
+func applyPanicDevicePreferences(preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
+	if len(preferenceSpec.Devices.PreferredPanicDevices) == 0 {
+		return
+	}
+
+	emptyPanicDevice := virtv1.PanicDevice{}
+	foundVMIEmptyPanicDevice := false
+	foundPreferredEmptyPanicDevice := false
+
+	// Only apply any preferred panic device when the same panic device has not been provided by a user already
+	panicDeviceModels := make(map[string]struct{})
+	for _, panicDevice := range vmiSpec.Domain.Devices.PanicDevices {
+		if panicDevice.Model == nil {
+			foundVMIEmptyPanicDevice = true
```

Comment:
Could you just use the following to represent an device without a model? Would remove the need for these `bools`:

```
const nilModel = "nilModel"
[..]
panicDeviceModels[nilModel] = struct{}{}
```

Reply:
Thanks. I've reworked that code with the hopefully non-conflicting "nilModel" string.